src/rust/engine/src/tasks.rs

```diff
@@ -8,9 +8,13 @@ use selectors::{Selector, Select, SelectDependencies, SelectLiteral, SelectProje
  * Registry of tasks able to produce each type, along with a few fundamental python
  * types that the engine must be aware of.
  */
+#[derive(Clone)]
```

Comment:
Can you open a github issue about removing `Clone` here? It's a bit error prone, because the tasks get passed around everywhere currently, and cloning them accidentally (probably in a parent struct) would get costly fast.